R/FilterState.R

```diff
@@ -284,6 +284,8 @@ FilterState <- R6::R6Class( # nolint
             lapply(private$observers, function(x) x$destroy())
           }
 
+          private$state_history <- reactiveVal(list())
```

Comment:
FYI, This ensures that when we call the server again (happens when teal tab is switched). The state history is clear, else we would get the `back` and `reset` buttons during init.

Screenshot when a new teal tab was clicked by the user and this was the state of the filter panel during init.
<img width="357" alt="Screenshot 2024-08-05 at 8 20 30 AM" src="https://github.com/user-attachments/assets/a847a96d-492e-492e-a28b-57595050149b">